ui/store/actions.ts

```diff
@@ -1844,8 +1843,10 @@ export function setSelectedAccount(
       getUnconnectedAccountAlertEnabledness(state);
     const activeTabOrigin = state.activeTab.origin;
     const internalAccount = getInternalAccountByAddress(state, address);
-    const permittedAccountsForCurrentTab =
-      getPermittedAccountsForCurrentTab(state);
+    // TODO: fix this
+    // const permittedAccountsForCurrentTab =
+    //   getPermittedAccountsForCurrentTab(state);
+    const permittedAccountsForCurrentTab = [];
```

Comment:
the logic in this helper is bugged in main. notice below how `currentTabIsConnectedToPreviousAddress` and `currentTabIsConnectedToNextAddress` effectively are checking the same address. Likely `internalAccount` should be  getting the account for the active tab instead